packages/esm-service-queues-app/src/add-provider-queue-room/add-provider-queue-room.component.tsx

```diff
@@ -148,75 +162,90 @@ const AddProviderQueueRoom: React.FC<AddProviderQueueRoomProps> = ({ closeModal,
       <ModalBody>
         <Form onSubmit={onSubmit}>
           <section className={styles.section}>
-            <Dropdown
-              aria-label={t('selectQueueLocation', 'Select a queue location')}
-              id="queueLocation"
-              initialSelectedItem={{ uuid: currentLocationUuid, name: currentLocationName }}
-              items={queueLocations}
-              itemToString={(item) => (item ? item.name : '')}
-              label=""
-              onChange={handleQueueLocationChange}
-              size="md"
-              titleText={t('queueLocation', 'Queue location')}
-              type="default"
+            <Controller
+              control={control}
+              name="queueLocationUuid"
+              render={({ field }) => (
+                <Dropdown
+                  {...field}
+                  aria-label={t('selectQueueLocation', 'Select a queue location')}
+                  id="queueLocation"
+                  initialSelectedItem={{ uuid: currentLocationUuid, name: currentLocationName }}
+                  items={queueLocations ?? []}
+                  itemToString={(item) => (item ? item.name : '')}
+                  onChange={(e) => {
+                    field.onChange(e.selectedItem?.id);
```

Comment:
```suggestion
                    if (!e.selectedItem) {
                      return;
                    }
                    field.onChange(e.selectedItem?.id);
```

We could add a null check to the handler here